src/misceval.js

```diff
@@ -514,55 +495,43 @@ Sk.misceval.richCompareBool = function (v, w, op, canSuspend) {
                 throw new Sk.builtin.TypeError("comparison did not return an int");
             }
         }
-
-    }
-    
-
-    
-    // handle special cases for comparing None with None or Bool with Bool
-    if (((v instanceof Sk.builtin.none) && (w instanceof Sk.builtin.none)) ||
-        ((v instanceof Sk.builtin.bool) && (w instanceof Sk.builtin.bool))) {
-        // Javascript happens to return the same values when comparing null
-        // with null or true/false with true/false as Python does when
-        // comparing None with None or True/False with True/False
-
-        if (op === "Eq") {
-            return v.v === w.v;
-        }
-        if (op === "NotEq") {
-            return v.v !== w.v;
-        }
-        if (op === "Gt") {
-            return v.v > w.v;
-        }
-        if (op === "GtE") {
-            return v.v >= w.v;
-        }
-        if (op === "Lt") {
-            return v.v < w.v;
-        }
-        if (op === "LtE") {
-            return v.v <= w.v;
+        // handle special cases for comparing None with None or Bool with Bool
+        if (v === Sk.builtin.none.none$ && w === Sk.builtin.none.none$) {
+            // Javascript happens to return the same values when comparing null
+            // with null or true/false with true/false as Python does when
+            // comparing None with None or True/False with True/False
+
+            if (op === "Eq") {
+                return v.v === w.v;
+            }
+            if (op === "NotEq") {
+                return v.v !== w.v;
+            }
+            if (op === "Gt") {
+                return v.v > w.v;
+            }
+            if (op === "GtE") {
+                return v.v >= w.v;
+            }
+            if (op === "Lt") {
+                return v.v < w.v;
+            }
+            if (op === "LtE") {
+                return v.v <= w.v;
+            }
         }
     }
 
-
     // handle equality comparisons for any remaining objects
     if (op === "Eq") {
-        if ((v instanceof Sk.builtin.str) && (w instanceof Sk.builtin.str)) {
-            return v.v === w.v;
-        }
```

Comment:
Why is it the case that this is no longer necessary? I believe that it can be cut, but I don't know/remember why it was here in the first place. With string interning working correctly, I agree it should not be necessary, but I'm sure this was here for a reason at some point in time.

Reply:
NumberCompare checks if other instanceof int_ which is redundant since we know already. So I didn't see the abstraction as beneficial with float or int here.

Reply:
I've gone through the blame.
interned - [Jul 2010](https://github.com/skulpt/skulpt/commit/720410b5aecba0d54dc84a6883895b95e138895e)
this code [Aug 2010](https://github.com/skulpt/skulpt/commit/55639a1ffa9281bd60ed18f7e3b4e87f362c180b)

fixes to interned much later...

So I presume this code was written as a reaction to `interned` not working completely... since it was written 1 month after interned was implemented and way before interned accounted for keys on `Object`'s `__proto__`. I think therefore it's safe to remove it now...

Reply:
Fair enough!